torch/csrc/utils/python_numbers.h

```diff
@@ -57,6 +57,29 @@ inline int64_t THPUtils_unpackLong(PyObject* obj) {
   return (int64_t)value;
 }
 
+inline bool THPUtils_checkIndex(PyObject *obj) {
+  if (THPUtils_checkLong(obj)) {
+    return true;
+  }
```

Comment:
I think this should have something like:

```
if (PyBool_Check(Obj)) return false
```

Reply:
sounds right.